docs/contributing/mac-deps-setup.md

```diff
@@ -0,0 +1,41 @@
+# macOS System Dependencies
+
+You will need to install these tools on your machine:
+
+ - Node.js v8.11.4
+ - Python 2.7
+ - Xcode and Xcode Command Line Tools (Xcode -> Preferences -> Downloads)
+
+## Node.js
+
+There are two approaches to installing Node.js, **manual** and **managed**. We recommend the managed route, but the simple method will work fine if you don't have node installed on your machine and just want to get running!
```

Comment:
> because people at WSC code day encountered the most problems with setup when they already had nodejs and/or python installed on their machine

Can you elaborate on what these problems looked like, for context? It's been a while since I've stumbled upon a problem like this with random Node codebases.

Reply:
mostly it was that they had a newer version of nodejs installed on their machine (and didn't want to uninstall it). the brew instructions may have helped some of them, but many wanted to keep their current version of node installed and work on our project too. i think part of it too was that `yarn install` failed for newer versions of node, but i can't remember for sure.

Reply:
> mostly it was that they had a newer version of nodejs installed on their machine (and didn't want to uninstall it).

I've had my eye on getting us up to Node 10 locally which will hopefully make this a non-issue. Last I checked it was just `node-sass` that started doing weird things when we upgraded it. If we can land that and no longer rely on Node 8 exclusively, would that change the sort of docs we need to write here?

Reply:
> If we can land that and no longer rely on Node 8 exclusively, would that change the sort of docs we need to write here?

https://github.com/desktop/desktop/pull/5644